lib/src/native/realm_core.dart

```diff
@@ -1665,35 +1669,33 @@ class _RealmCore {
     });
   }
 
-  static void _logCallback(Object userdata, int levelAsInt, Pointer<Int8> message) {
-    final logger = Realm.logger;
-    final level = LevelExt.fromInt(levelAsInt);
+  @override
+  void logMessage(int level, String message) {
+    final message_level = LevelExt.fromInt(level);
+    Realm.logger.log(message_level, message);
+  }
 
-    // Don't do expensive utf8 to utf16 conversion unless needed.
-    if (logger.isLoggable(level)) {
-      logger.log(level, message.cast<Utf8>().toDartString());
+  void _initDefaultLogger() {
+    bool isDefaultLogger = _realmLib.realm_dart_init_default_logger(RealmInternal.defaultLogger.level.toInt());
+    if (isDefaultLogger && !RealmInternal.isUserLogger) {
```

Comment:
isUserLogger is seems not needed here. 
It is either isDefaultLogger true or not.